service/src/main/java/greencity/service/GoogleApiService.java

```diff
@@ -80,16 +80,18 @@ private AddressResponse getAddressResponseByLocaleAndCoordinates(LatLng latLng,
         try {
             GeocodingResult[] results = GeocodingApi.newRequest(context)
                 .latlng(latLng).language(locale.getLanguage()).await();
-            return getAddressResponse(results[0].addressComponents);
+            return getAddressResponse(results[0]);
```

Comment:
Can we have the `IndexOutOfBound` exception here?

Reply:
No
If the geocoding was successful but returned no results, the ApiException (ZeroResultsException) is thrown and caught below.